extensions/jackson/src/test/groovy/io/jsonwebtoken/jackson/io/JacksonDeserializerTest.groovy

```diff
@@ -105,6 +105,41 @@ class JacksonDeserializerTest {
         assertEquals expected, result
     }
 
+    /**
+     * For: https://github.com/jwtk/jjwt/issues/564
+     */
+    @Test
+    void testMappedTypeDeserializerWithMapNullCheck() {
+
+        // mimic map implementations that do NOT allow for null keys, or containsKey(null)
+        Map typeMap = new HashMap() {
+            @Override
+            boolean containsKey(Object key) {
+                if (key == null) {
+                    throw new NullPointerException("key is null, expected for this test")
+                }
+                return super.containsKey(key)
+            }
+        }
+
+        // TODO: the following does NOT work with Java 1.7
+        // when we stop supporting that version we can use a partial mock instead
+        // the `typeMap.put("custom", CustomBean)` line below results in an NPE, (only on 1.7)
+
```

Comment:
@lhazlewood any ideas on how to clean up this test?
The partial mock doesn't work on 1.7, and I'm not sure I want to dig into that, but I feel like the above solution could be better too...

Reply:
I don't know about the `partialMockBuilder` myself - never used it.  That said, I like creating the anonymous subclass - that's even better than a mock IMO.